frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/DropDownSelect.jsx

```diff
@@ -0,0 +1,219 @@
+        <ButtonSolid
+          size="sm"
+          variant="tertiary"
+          disabled={disabled}
+          onClick={(e) => {
+            e.stopPropagation();
+            if (disabled) {
+              return;
+            }
+            setShowMenu((show) => !show);
+          }}
+          className="tdb-dropdown-btn px-1 pe-3 ps-2 gap-0 w-100 border-0 justify-content-start rounded-0 position-relative font-weight-normal"
+          data-cy={`show-ds-popover-button`}
+        >
+          {renderSelected && renderSelected(selected)}
+          {!renderSelected && selected
+            ? Array.isArray(selected)
+              ? !isOverflown && (
+                  <MultiSelectValueBadge
+                    options={options}
+                    selected={selected}
+                    setSelected={setSelected}
+                    onChange={onChange}
+                  />
+                )
+              : selected?.label
+            : ''}
+          {!renderSelected && isOverflown && !Array.isArray(selected) && (
+            <Badge className="me-1 dd-select-value-badge" bg="secondary">
+              {selected?.length} selected
+              <span
+                role="button"
+                onClick={(e) => {
+                  setSelected([]);
+                  onChange && onChange([]);
+                  e.preventDefault();
+                  e.stopPropagation();
+                }}
+              >
+                <Remove fill="var(--slate12)" />
+              </span>
+            </Badge>
+          )}
+          <span className="dd-select-control-chevron">
+            <CheveronDown />
+          </span>
+        </ButtonSolid>
+      </span>
+    </OverlayTrigger>
+  );
+};
+
+function MultiSelectValueBadge({ options, selected, setSelected, onChange }) {
+  if (options?.length === selected?.length && selected?.length !== 0) {
+    return (
+      <Badge className={`me-1 dd-select-value-badge`} bg="secondary">
+        All {options?.length} selected
+        <span
+          role="button"
+          onClick={(e) => {
+            e.stopPropagation();
+            setSelected([]);
+            onChange([]);
+            e.preventDefault();
+          }}
+        >
+          <Remove fill="var(--slate12)" />
+        </span>
+      </Badge>
+    );
+  }
+
+  return selected.map((option) => (
+    <Badge key={option.value} className="me-1 dd-select-value-badge" bg="secondary">
+      {option.label}
+      <span
+        role="button"
+        onClick={(e) => {
+          setSelected((selected) => {
+            onChange && onChange(selected.filter((opt) => opt.value !== option.value));
+            return selected.filter((opt) => opt.value !== option.value);
+          });
+          e.preventDefault();
+          e.stopPropagation();
+        }}
+      >
+        <Remove fill="var(--slate12)" />
+      </span>
+    </Badge>
+  ));
+}
+
+function generateRandomId(length) {
```

Comment:
note, you can rather use `uuidv4` rather

Reply:
Added in upcoming PR